data/balance.go

```diff
@@ -79,19 +85,48 @@ func (txm *TransactionWithMetaData) Balances() (BalanceMap, error) {
 			switch node.CreatedNode.LedgerEntryType {
 			case ACCOUNT_ROOT:
 				created := node.CreatedNode.NewFields.(*AccountRoot)
-				balanceMap.Add(created.Account, &zeroAccount, &zeroNative, created.Balance, &zeroCurrency)
+				balanceMap.Add(created.Account, &zeroAccount, created.Balance, created.Balance, &zeroCurrency, created.AMMID)
 			case RIPPLE_STATE:
 				// New trust line
 				state := node.CreatedNode.NewFields.(*RippleState)
-				balanceMap.Add(&state.LowLimit.Issuer, &state.HighLimit.Issuer, state.Balance.Value, state.Balance.Value, &state.Balance.Currency)
-				balanceMap.Add(&state.HighLimit.Issuer, &state.LowLimit.Issuer, state.Balance.Value.Negate(), state.Balance.Value.Negate(), &state.Balance.Currency)
+				balanceMap.Add(&state.LowLimit.Issuer, &state.HighLimit.Issuer, state.Balance.Value, state.Balance.Value, &state.Balance.Currency, nil)
+				balanceMap.Add(&state.HighLimit.Issuer, &state.LowLimit.Issuer, state.Balance.Value.Negate(), state.Balance.Value.Negate(), &state.Balance.Currency, nil)
+			case AMMROOT:
+				state := node.CreatedNode.NewFields.(*AMM)
+				balanceMap.Add(&state.LPTokenBalance.Issuer,
+					&state.LPTokenBalance.Issuer,
+					state.LPTokenBalance.Value,
+					state.LPTokenBalance.Value,
+					&state.LPTokenBalance.Currency,
+					nil)
+
 			}
 		case node.DeletedNode != nil:
 			switch node.DeletedNode.LedgerEntryType {
 			case RIPPLE_STATE:
-				//?
+				// A deletion (complete termination of a token) can lead to having to use
+				// the deleted node to determine the balance of the counterparty.
+				var previous, current *RippleState
+				if node.DeletedNode.PreviousFields == nil || node.DeletedNode.FinalFields == nil {
+					continue
+				}
+				previous = node.DeletedNode.PreviousFields.(*RippleState)
+				current = node.DeletedNode.FinalFields.(*RippleState)
+
+				if previous.Balance == nil {
+					//flag change
+					continue
+				}
+				change, err := current.Balance.Subtract(previous.Balance)
+				if err != nil {
+					return nil, err
+				}
+				balanceMap.Add(&current.LowLimit.Issuer, &current.HighLimit.Issuer, current.Balance.Value, change.Value, &current.Balance.Currency, nil)
+				balanceMap.Add(&current.HighLimit.Issuer, &current.LowLimit.Issuer, current.Balance.Value.Negate(), change.Value.Negate(), &current.Balance.Currency, nil)
 			case ACCOUNT_ROOT:
 				return nil, fmt.Errorf("Deleted AccountRoot!")
+			case AMMROOT:
+				// 20230911, NvN: Looks like it is not required to handle this case
```

Comment:
Putting a date and initials in a comment is bad practice.

Reply:
It is a note to self :)